polls/views.py

```diff
@@ -45,25 +45,35 @@ def get(self, request):
         can_continue, should_rest, _ = check_user_work_permission(request.user)
         if should_rest:
             return redirect("auth-flow")
-        elif can_continue:
+
+        if can_continue:
             # Update user's session start time
             request.user.first_task_of_this_session_performed_at = timezone.now()
             request.user.save()
 
+        current_batch = ""
         if batch_selector():
-            # when probability lesss than 90%
-            current_batch = CurrentBatchEval.objects.first().current_batch_eval
+            current_batch_eval = CurrentBatchEval.objects.first()
+            current_batch = (
+                current_batch_eval.current_batch_eval if current_batch_eval else ""
+            )
         else:
-            current_batch = CurrentBatchGold.objects.first().current_batch_gold
-        all_tasks = current_batch.tasks.all()
-        tasks_for_user = all_tasks.exclude(annotation__user=request.user)
-        task = tasks_for_user.first()
-        context = {"task": task}
-        if not task:
-            return render(request, self.template_name, context)
-        url, task_presentation = present_task_for_user(task)
-        context["url"] = url
-        context["task_presentation"] = task_presentation
+            current_batch_gold = CurrentBatchGold.objects.first()
+            current_batch = (
+                current_batch_gold.current_batch_gold if current_batch_gold else ""
+            )
+
+        context = {}
+        if current_batch:
+            tasks_for_user = current_batch.tasks.exclude(annotation__user=request.user)
+            task = tasks_for_user.first()
+            if task:
+                url, task_presentation = present_task_for_user(task)
+                context = {
+                    "task": task,
+                    "url": url,
+                    "task_presentation": task_presentation,
+                }
         return render(request, self.template_name, context)
 
```

Comment:
Great code clean-up, would you mind adding a little unit test to assert that this still works as expected? If your answer is negative kindly let me know. @najamF18

Reply:
Thank you. Sure I will add a unit test for this too

Reply:
> Thank you. Sure I will add a unit test for this too

@DeFidelity we can add a test case for this but this will redirect it to login url because we need to authenticate using allauth. We can add the asserts to see if it is being redirected to correct url. Should we go ahead with this? Awaiting your response

Reply:
Okay, great. 
I think it's good for a merge then

Reply:
> Okay, great. I think it's good for a merge then

Great. Do we need to add this test to see if redirection is correct?

Reply:
@najamF18 absolutely No, we have some more important specs to work on, I was told to create new issues for additional functionalities, I'll appreciate if you can check the log for that test that fails in the other PR, the test passed before maybe something changed by mistake, two previously passed test are now failing and that's what caused the error.
At times all that needs to be done is just repushing the code if nothing changed in the test file from the previous one that passed